lib/spanner/config/parser.ex

```diff
@@ -0,0 +1,53 @@
+defmodule Spanner.Config.Parser do
+
+  @moduledoc """
+  Functions for parsing YAML config files
+  """
+
+  @doc """
+  Reads a YAML file from 'path'. The underlying lib, yamerl, throws when
+  errors occur. We catch those and return tuple containing the atom ':error'
+  and the list error messages.
+  """
+  def read_from_file(path) do
+    try do
+      yaml = YamlElixir.read_from_file(path)
+      # We should never get an empty map back from 'YamlElixir.read_from_file/1'.
+      # If we do, we return an error.
+      if length(Map.values(yaml)) == 0 do
+        {:error, ["Empty map returned. Make sure there are no errors in your YAML."]}
```

Comment:
What are the types of errors which would make `YamlElixir` return an empty map without throwing parse errors?